sound/soc/sof/sof-acpi-dev.c

```diff
@@ -93,21 +93,16 @@ int sof_acpi_probe(struct platform_device *pdev, const struct sof_dev_desc *desc
 		sof_pdata->tplg_filename_prefix =
 			sof_pdata->desc->default_tplg_path;
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_PROBE_WORK_QUEUE)
-	/* set callback to enable runtime_pm */
+	/* set callback to be called on successful device probe to enable runtime_pm */
 	sof_pdata->sof_probe_complete = sof_acpi_probe_complete;
-#endif
+
```

Comment:
Humm, good topic that results in many questions on my side. Thanks @ujfalusi 

a)
the core will use this:

````c
	if (plat_data->sof_probe_complete)
		plat_data->sof_probe_complete(sdev->dev);
````

Are we now saying that we require this function to be set?

b) I am positive that in the case of ACPI devices alone, we did not set the workqueue option (I added an option to force this recently), so as a result we never called probe_complete. It turns out that this was harmless, because in practice we don't have pm_runtime support for ACPI devices. 

c) I am not sure what happened in the nocodec case as well, since it's my understanding we don't have the workqueue option set, so wondering now if pm_runtime was ever enabled on PCI device w/ nocodec?

Reply:
@ujfalusi did you crash the CI with your first patch? welcome to SOF! But what happens when you enable runtime PM before finishing the workqueue? I think the specific reason we added this change was to prevent the SOF PCI device from enabling runtime PM unless the DSP probe was also successful (which is done in the workqueue)

Reply:
@ranj063 , nothing changes with this patch, it is just removing the ifdefery from the acpi/pci/of code. The end result is exactly the same.
If we use the work, then the sof_TYPE_probe_complete() will be called from the work.
If we don't use the work then the sof_TYPE_probe_complete() will be called at the same time as it used to be.

But it will ensure that the software flow will be the same for all cases and removes duplicated 'logic' from the acpi/pci/of drivers.

Reply:
> If we use the work, then the sof_TYPE_probe_complete() will be called from the work.

agree

> If we don't use the work then the sof_TYPE_probe_complete() will be called at the same time as it used to be.

we didn't set the callback before, so that's a change in behavior.